ocaml/libs/tracing/tracing.ml

```diff
@@ -634,3 +634,34 @@ let with_tracing ?(attributes = []) ?(parent = None) ~name f =
     | Error e ->
         warn "Failed to start tracing: %s" (Printexc.to_string e) ;
         f None
+
+module EnvHelpers = struct
+  let traceparent_key = "TRACEPARENT"
+
+  let of_traceparent traceparent =
+    match traceparent with
+    | None ->
+        []
+    | Some traceparent ->
+        [String.concat "=" [traceparent_key; traceparent]]
+
+  let to_traceparent env =
+    let env_opt =
+      List.find_opt (String.starts_with ~prefix:traceparent_key) env
+    in
+    Option.bind env_opt (fun key_value ->
+        match String.split_on_char '=' key_value with
+        | ["TRACEPARENT"; traceparent] ->
```

Comment:
traceparent_key here ?

Reply:
I cannot use pattern matching for strings with the name of the binding. If I use  `[traceparent_key ; traceparent]` it will match to any list of 2 strings and `traceparent_key` becomes the first entry in the list for the scope of the match.

Reply:
```
| [key; traceparent] when key = "TRACEPARENT" ->
```
?

Reply:
Sure!

Reply:
@mg12 https://github.com/xapi-project/xen-api/pull/5583/commits/45eeca79eabd49dad52b8778e4d95f6a908a6028 here.